provider/pkg/await/deployment.go

```diff
@@ -144,53 +144,72 @@ func (dia *deploymentInitAwaiter) Await() error {
 	//      because it doesn't do a rollout (i.e., it simply creates the Deployment and
 	//      corresponding ReplicaSet), and therefore there is no rollout to mark as "Progressing".
 	//
-
-	deploymentClient, replicaSetClient, podClient, pvcClient, err := dia.makeClients()
-	if err != nil {
-		return err
-	}
-
-	// Create Deployment watcher.
-	deploymentWatcher, err := deploymentClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "could not set up watch for Deployment object %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer deploymentWatcher.Stop()
-
-	// Create ReplicaSet watcher.
-	replicaSetWatcher, err := replicaSetClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for ReplicaSet objects associated with Deployment %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer replicaSetWatcher.Stop()
-
-	// Create Pod watcher.
-	podWatcher, err := podClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for Pods objects associated with Deployment %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer podWatcher.Stop()
-
-	// Create PersistentVolumeClaims watcher.
-	pvcWatcher, err := pvcClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for PersistentVolumeClaims objects associated with Deployment %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer pvcWatcher.Stop()
+	stopper := make(chan struct{})
+	defer close(stopper)
+
+	// Limit the lifetime of this to each deployment await for now. We can reduce this sharing further later.
```

Comment:
Any particular reason for initializing a factory per await? Seems like we could share the factory across the provider without much difference in level of effort.

Reply:
> Any particular reason for initializing a factory per await? Seems like we could share the factory across the provider without much difference in level of effort.

One thing was to set the lifetime of the factory itself (the stop channel to pass to the `Start` call in the next line). This way we know it's safe to kill the informer once an individual deployment has been waited on. A shared informer for the entire provider would need some degree of coordination to make sure all the deployments had completed etc. which I didn't want to layer on yet.

Reply:
Ok. That sounds reasonable as a first step, but presumably doesn't buy us anything in terms of performance since we aren't sharing the cache between resources?